### ISSUE TITLE
Problem with UTF-8 file/directory names

### ISSUE THREAD

**Issue by zaak** [thephpleague/flysystem#235]
I have following directory structure:

```
.
└── files
    ├── 繁體中文字
    │   └── test.txt
    └── 繁體中文字.txt
```

When I call `$filesystem->listContents('files')` I'm getting following output:

```
array (size=2)
  0 => 
    array (size=6)
      'dirname' => string 'files' (length=5)
      'basename' => string 'files' (length=5)
      'filename' => string 'files' (length=5)
      'path' => string 'files/繁體中文字' (length=21)
      'type' => string 'dir' (length=3)
      'timestamp' => int 1409223899
  1 => 
    array (size=8)
      'dirname' => string 'files' (length=5)
      'basename' => string '.txt' (length=4)
      'extension' => string 'txt' (length=3)
      'filename' => string '' (length=0)
      'path' => string 'files/繁體中文字.txt' (length=25)
      'type' => string 'file' (length=4)
      'timestamp' => int 1409223047
      'size' => int 0
```

As you can see `filename` and `basename` are invalid. This is caused by lack of multibyte support in `pathinfo` used in [Util::pathinfo()](https://github.com/thephpleague/flysystem/blob/master/src/Util.php#L18). Expected output is:

```
array (size=2)
  0 => 
    array (size=7)
      'dirname' => string 'files' (length=5)
      'basename' => string '繁體中文字' (length=15)
      'extension' => string '' (length=0)
      'filename' => string '繁體中文字' (length=15)
      'path' => string 'files/繁體中文字' (length=21)
      'type' => string 'dir' (length=3)
      'timestamp' => int 1409223899
  1 => 
    array (size=8)
      'dirname' => string 'files' (length=5)
      'basename' => string '繁體中文字.txt' (length=19)
      'extension' => string 'txt' (length=3)
      'filename' => string '繁體中文字' (length=15)
      'path' => string 'files/繁體中文字.txt' (length=25)
      'type' => string 'file' (length=4)
      'timestamp' => int 1409223047
      'size' => int 0
```

**Comment by ThaDafinser** [comment]
@zaak did you had this issue under windows or unix?

**Comment by zaak** [comment]
@ThaDafinser Linux. On Windows it fails completely if it's about UTF-8 names, but it's PHP itself at fault, not the Flysystem.

**Comment by ThaDafinser** [comment]
@zaak thank you.

Just tried on PHP7 and still not solved.

I found this for windows (in case someone comes around here):
https://github.com/kenjiuno/php-wfio

Alternative with ugly `com` extension:
```php
<?php
$path = '..YOUR_PATH...';

/*
 * COM
 */
$fso = new \COM('Scripting.FileSystemObject', null, CP_UTF8);

$folder = $fso->GetFolder($path);

foreach ($folder->Files as $obj) {
    var_dump($obj->Name);
}
```

**Comment by adrianopedro** [comment]
I also found this problem...
The problem is that at the origin the files are url encoded for safety, but then are not reverted.
In the pathinfo, just treat the filename with urldecode.

``` 
public static function pathinfo($path)
    {

       $pathinfo = pathinfo($path) + compact('path');

        //Fix problem with filenames having spacing or special chars 
        //that are url encoded for safety, but then are not reverted.
        $pathinfo['filename'] = urldecode($pathinfo['filename']);

        $pathinfo['dirname'] = array_key_exists('dirname', $pathinfo)
            ? static::normalizeDirname($pathinfo['dirname'])
            : '';

        return $pathinfo;
    }
```

**Comment by ThaDafinser** [comment]
Most or all of the issues here should be resolved with php 7.1
https://github.com/php/php-src/blob/PHP-7.1.0beta2/UPGRADING#L321

**Comment by shinnlu** [comment]
i have the issue with php 7.3.9
![image](https://user-images.githubusercontent.com/570246/65844099-65c42200-e367-11e9-9847-6d6cff93285d.png)
the issue was on Ubuntu, but working fine on MacOS.
